### PR TITLE
chore(core): upgrade to v0.49.0

### PR DESCRIPTION
**Pull request by sumo-drosiek**
Upgrade core to [v0.49.0](https://github.com/open-telemetry/opentelemetry-collector-contrib/releases/tag/v0.49.0)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,13 +9,24 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Changed
 
+- chore: bump OT core to v0.49.0 [#550][#550]
+
+[Unreleased]: https://github.com/SumoLogic/sumologic-otel-collector/compare/v0.48.0-sumo-0...main
+[#550]: https://github.com/SumoLogic/sumologic-otel-collector/pull/550
+
+## [v0.48.0-sumo-0]
```
Comment (pmalek-sumo) [inline CHANGELOG.md:17]: :)

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,13 +9,24 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+### Changed
+
 - chore: bump OT core to v0.48.0 [#534][#534]
 
 ### Fixed
 
 - fix(sumologicexporter): treat resource attributes as fields for otlp #536
 
-[Unreleased]: https://github.com/SumoLogic/sumologic-otel-collector/compare/v0.47.0-sumo-0...main
+[v0.48.0-sumo-0]: https://github.com/SumoLogic/sumologic-otel-collector/compare/v0.47.0-sumo-0...v0.47.0-sumo-0
```
Comment (pmalek-sumo) [inline CHANGELOG.md:29]:
```suggestion
[v0.48.0-sumo-0]: https://github.com/SumoLogic/sumologic-otel-collector/compare/v0.47.0-sumo-0...v0.48.0-sumo-0
```

File: Makefile [hunk 3]
```diff
@@ -85,13 +85,13 @@ OT_CORE_VERSION := $(shell grep "otelcol_version: .*" otelcolbuilder/.otelcol-bu
...
 	@test $(OT_CORE_NEW_VERSION) || (echo "usage: make update-otc-core OT_CORE_NEW_VERSION=x.x.x"; exit 1);
 	@echo "updating OT core from $(OT_CORE_VERSION) to $(OT_CORE_NEW_VERSION)"
-	@sed -i "" "s/$(OT_CORE_VERSION)/$(OT_CORE_NEW_VERSION)/" otelcolbuilder/.otelcol-builder.yaml
-	@sed -i "" "s/$(OT_CORE_VERSION)/$(OT_CORE_NEW_VERSION)/" otelcolbuilder/Makefile
-	@sed -i "" "s/$(OT_CORE_VERSION)/$(OT_CORE_NEW_VERSION)/" README.md
-	@sed -i "" "s/$(OT_CORE_VERSION)/$(OT_CORE_NEW_VERSION)/" docs/Configuration.md
-	@sed -i "" "s/$(OT_CORE_VERSION)/$(OT_CORE_NEW_VERSION)/" docs/KnownIssues.md
-	@sed -i "" "s/$(OT_CORE_VERSION)/$(OT_CORE_NEW_VERSION)/" pkg/receiver/telegrafreceiver/README.md
-	@find . -type f -name "go.mod" -exec sed -i "" "s/$(OT_CORE_VERSION)/$(OT_CORE_NEW_VERSION)/" {} \;
+	sed -i'' -e "s/$(OT_CORE_VERSION)/$(OT_CORE_NEW_VERSION)/" otelcolbuilder/.otelcol-builder.yaml
```
Comment (pmalek-sumo) [inline Makefile:88]: uuuu gnu vs mac `sed` 🤯 

Perhaps we'd need to adjust the `Makefile` to work with both?

Comment (swiatekm-sumo) [reply Makefile:88]: According to [this answer](https://unix.stackexchange.com/questions/13711/differences-between-sed-on-mac-osx-and-other-standard-sed) `-i -e` should work on both. Considering the number of differences though, I wonder if we shouldn't use a different utility. `awk` perhaps?

Comment (sumo-drosiek) [reply Makefile]: I used gsed approach, which has been used for us previously in this Makefile

**Review by swiatekm-sumo — APPROVED** [review]

**Pull request merged**